File: static/code/sample/tethered/gdl-acrelayx1-a.py
# In this sample:
# Continuously switch relay on/off every 2 seconds

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def Set(value):
    v = 1 if value else 0
    duelink.Engine.ExecuteCommand(f"Set({v})")

while True:
    time.sleep(2)
    Set(True)

    time.sleep(2)
    Set(False)

File: static/code/sample/tethered/gdl-acrelayx1-a.cs
// In this sample:
// Continuously switch relay on/off every 2 seconds

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void Set(bool value) {
    var v = value ? 1 : 0;
    duelink.Engine.ExecuteCommand($"Set({v})");
}

while (true) {
    Thread.Sleep(2000);
    Set(true);

    Thread.Sleep(2000);
    Set(false);
}
